### DISCUSSION TITLE
how to call the list of players in the town and check the kick, exit from the town?

### DISCUSSION THREAD

**Discussion by Ru5D1mG4me** [TownyAdvanced/Towny#4999] [Q&A] [answered]
Hello, I am writing a plugin for passports, but I am facing a problem. I re-read all the API documentation, but I couldn't find anything for my problem. How to find out in the code which players are in the town(list)? And one more question how to track that the player left the town or was kicked?

**Comment by LlmDl** [comment, marked as answer]
You can use town.getResidents() to get the players who belong to a town.

The TownLeaveEvent and TownKickEvent are events Towny throws to let you know when players are losing their Town.

**Reply by Ru5D1mG4me** [reply]
thank you
